### DISCUSSION TITLE
Gradle lockfile to CycloneDX: Fill in "component.group"

### DISCUSSION THREAD

**Discussion by ChristianCiach** [aquasecurity/trivy#5970] [Bugs]
### Description

We are using Trivy to generate a CycloneDX SBOM from a `gradle.lockfile`:

```
trivy fs gradle.lockfile --format cyclonedx --output sbom.cdx.json
```

This works, but Trivy actually generates an incorrect SBOM. According to the CycloneDX specification, the `group` attribute of a `component` should state the maven group of the component. Instead, Trivy writes *both* the maven-group and the maven-name into the `name` attribute of the `component`.

Interestingly, this bug did also exist when scanning jar-files, but this been fixed quite a while ago: https://github.com/aquasecurity/trivy/issues/4675 We need to do the same thing here.

### Desired Behavior

Let's assume the `gradle.lockfile` contains this entry:

```
com.google.guava:guava:32.1.1-jre=runtimeClasspath
```

When using Trivy to convert the above to CycloneDX, the component *should* be included like this:

```json
    {
      "bom-ref": "pkg:maven/com.google.guava/guava@32.1.1-jre",
      "type": "library",
      "group": "com.google.guava",
      "name": "guava",
      "version": "32.1.1-jre",
      "purl": "pkg:maven/com.google.guava/guava@32.1.1-jre",
      "properties": [
        {
          "name": "aquasecurity:trivy:PkgType",
          "value": "gradle"
        }
      ]
    }
```

### Actual Behavior

Unfortunately, it actually will look like this:

```json
    {
      "bom-ref": "pkg:maven/com.google.guava/guava@32.1.1-jre",
      "type": "library",
      "name": "com.google.guava:guava",
      "version": "32.1.1-jre",
      "purl": "pkg:maven/com.google.guava/guava@32.1.1-jre",
      "properties": [
        {
          "name": "aquasecurity:trivy:PkgType",
          "value": "gradle"
        }
      ]
    }
```

According to specs, this is an invalid `name`, because it should not include the maven-group. Instead, the maven-group should be included using the `group` attribute.

In other words, Trivy should just split the current `"name": "com.google.guava:guava"` into `"name": "guava"` and `"group": "com.google.guava"`. Many tools like https://github.com/DependencyTrack/dependency-track need this to work correctly.

### Reproduction Steps

1. Configure a Gradle project (`build.gradle`) for dependency locking: `dependencyLocking { lockAllConfigurations() }`
2. Generate the lockfile: `$ ./gradlew dependencies --write-locks`
3. Use trivy to convert the lockfile to cyclonedx: `$ trivy fs gradle.lockfile --format cyclonedx --output sbom.cdx.json`
4. Observe the components-array inside the generated file `sbom.cdx.json` to see the wrong `name` and the missing `group` attributes.

### Target

Filesystem

### Scanner

None

### Output Format

CycloneDX

### Mode

Standalone

### Debug Output

```bash
2024-01-18T20:58:01.425+0100    DEBUG   ["cyclonedx" "spdx" "spdx-json" "github"] automatically enables '--list-all-pkgs'.
2024-01-18T20:58:01.425+0100    DEBUG   Severities: ["UNKNOWN" "LOW" "MEDIUM" "HIGH" "CRITICAL"]
2024-01-18T20:58:01.425+0100    DEBUG   Ignore statuses {"statuses": null}
2024-01-18T20:58:01.426+0100    INFO    "--format cyclonedx" disables security scanning. Specify "--scanners vuln" explicitly if you want to include vulnerabilities in the CycloneDX report.
2024-01-18T20:58:01.434+0100    DEBUG   cache dir:  /home/christianc/.cache/trivy
2024-01-18T20:58:01.434+0100    DEBUG   Enabling misconfiguration scanners: [azure-arm cloudformation dockerfile helm kubernetes terraform terraformplan]
2024-01-18T20:58:01.434+0100    DEBUG   The nuget packages directory couldn't be found. License search disabled
2024-01-18T20:58:01.434+0100    DEBUG   Walk the file tree rooted at 'gradle.lockfile' in parallel
2024-01-18T20:58:01.435+0100    DEBUG   OS is not detected.
```

### Operating System

Fedora 39

### Version

```bash
Version: 0.48.3
```

### Checklist

- [X] Run `trivy image --reset`
- [X] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)

**Comment by DmitriyLewen** [comment]
Hello @ChristianCiach 
Thanks for your report!

We have already merged fix for this issue - https://github.com/aquasecurity/trivy/pull/5922.
It will be included in the next release.

Regards, Dmitriy
